### ISSUE TITLE
Backlog queue for tickets

### ISSUE THREAD

**Issue by Jzir** [chamburr/modmail#150]
**Please describe the feature you'd like**
I am a community manager for a very large discord, and we regularly reach the max channels per category limit, meaning that users cannot open new tickets, and get frustrated at this. It might be useful to have a 'ticket backlog' such that if the max tickets open is reached, any new users trying to open tickets go on a 'backlog' so that when one ticket is closed, the next ticket on the backlog is opened automatically. The bot would inform the user what position in the queue they're in and that they've been placed in the queue.

**Describe the motivation for the feature**
Users would spam the bot less, would feel more that they're being looked after and getting tickets in at busy times would be less RNG.

**Describe alternatives you've considered**
We've tried hacking our way around the 50 channels limit by making another bot of ours move modmail channels into a different category, basically giving us ~100 tickets open - this just gets filled as well and the mods get demotivated as we have a gigantic wall of tickets.

**Additional context**
Would need a toggle on/off for this feature, potentially a way for mods to look at the queue and remove people from it/block people from clogging it up due to misuse. Perhaps a rough waiting time given an estimation over the last X minutes/hours of tickets per hour, but obviously with the caveat that it's an estimation and not a promise - perhaps this message could be customised too.

**Comment by dlsnyder8** [comment]
It would be easier to implement a way to create overflow categories that new tickets automatically go into. 

Creating a queue system would not only be harder to implement, but would be hard to manage using Discord's interface, necessitating a dashboard of some kind.

**Comment by Scyye** [comment]
@dlsnyder8 I agree with the first part, however what about this would necessitate a dashboard?
